scipy/linalg/_decomp_update.pyx

```diff
@@ -0,0 +1,2416 @@
+                p_full = p1 - p_eco
+            if typecode == cnp.NPY_FLOAT:
+                info = thin_qr_row_delete(m, n, <float*>extract(q1, qs), qs, qisF,
+                    <float*>extract(r1, rs), rs, k1, p_eco, p_full)
+            elif typecode == cnp.NPY_DOUBLE:
+                info = thin_qr_row_delete(m, n, <double*>extract(q1, qs), qs, qisF,
+                    <double*>extract(r1, rs), rs, k1, p_eco, p_full)
+            elif typecode == cnp.NPY_CFLOAT:
+                info = thin_qr_row_delete(m, n, <float_complex*>extract(q1, qs), qs, qisF,
+                    <float_complex*>extract(r1, rs), rs, k1, p_eco, p_full)
+            else:  #  cnp.NPY_CDOUBLE
+                info = thin_qr_row_delete(m, n, <double_complex*>extract(q1, qs), qs, qisF,
+                    <double_complex*>extract(r1, rs), rs, k1, p_eco, p_full)
+            if info == 1:
+                return q1[p_full:-p_eco, p_full:], r1[p_full:,:]
+            elif info == MEMORY_ERROR:
+                raise MemoryError('malloc failed')
+            else:
+                raise ValueError('Reorthogonalization Failed, unable to perform row deletion.')
+        else:
+            if typecode == cnp.NPY_FLOAT:
+                qr_block_row_delete(m, n, <float*>extract(q1, qs), qs,
+                    <float*>extract(r1, rs), rs, k1, p1)
+            elif typecode == cnp.NPY_DOUBLE:
+                qr_block_row_delete(m, n, <double*>extract(q1, qs), qs,
+                    <double*>extract(r1, rs), rs, k1, p1)
+            elif typecode == cnp.NPY_CFLOAT:
+                qr_block_row_delete(m, n, <float_complex*>extract(q1, qs), qs,
+                    <float_complex*>extract(r1, rs), rs, k1, p1)
+            else:  # cnp.NPY_CDOUBLE:
+                qr_block_row_delete(m, n, <double_complex*>extract(q1, qs), qs,
+                    <double_complex*>extract(r1, rs), rs, k1, p1)
+            return q1[p1:, p1:], r1[p1:, :]
+    elif which == 'col':
+        if p1 > 1:
+            q1, r1, typecode, m, n, economic = validate_qr(Q, R, overwrite_qr,
+                    cnp.NPY_F_CONTIGUOUS, overwrite_qr, cnp.NPY_F_CONTIGUOUS,
+                    check_finite)
+        else:
+            q1, r1, typecode, m, n, economic = validate_qr(Q, R, overwrite_qr,
+                    NPY_ANYORDER, overwrite_qr, NPY_ANYORDER, check_finite)
+        o = n if economic else m
+        if not (-n <= k1 < n):
+            raise ValueError('k is not a valid index')
+        if k1 < 0:
+            k1 += n
+        if k1 + p1 > n or p1 <= 0:
+            raise ValueError('p out of range')
+
+        if p1 == 1:
+            if typecode == cnp.NPY_FLOAT:
+                qr_col_delete(m, o, n, <float*>extract(q1, qs), qs, 
+                    <float*>extract(r1, rs), rs, k1)
+            elif typecode == cnp.NPY_DOUBLE:
+                qr_col_delete(m, o, n, <double*>extract(q1, qs), qs, 
+                    <double*>extract(r1, rs), rs, k1)
+            el
```

Comment:
Cython's numpy support doesn't have these names.  I could add them to the top of this file and use them if you'd like, but it will likely still warn when compiling.